test/sub-menu.html

```diff
@@ -88,21 +89,106 @@
       });
 
       (isIOS ? it.skip : it)('should open sub-menu on arrow down', async() => {
+        const spy = sinon.spy();
+        menu._container.addEventListener('keydown', spy);
         arrowDown(buttons[0]);
         await nextRender(subMenu);
         expect(subMenu.opened).to.be.true;
+        expect(spy.firstCall.args[0].defaultPrevented).to.be.true;
       });
 
-      (isIOS ? it.skip : it)('should close sub-menu on first item arrow up', async() => {
-        arrowDown(buttons[0]);
+      (isIOS ? it.skip : it)('should open sub-menu and focus last item on arrow up', async() => {
+        arrowUp(buttons[0]);
+        await onceOpened(subMenu);
+        expect(subMenu.opened).to.be.true;
+        const items = subMenu.$.overlay.querySelectorAll('vaadin-context-menu-item');
+        const last = items[items.length - 1];
+        const spy = sinon.spy(last, 'focus');
         await nextRender(subMenu);
+        expect(spy).to.be.calledOnce;
+      });
 
+      (isIOS ? it.skip : it)('should close sub-menu on first item arrow up', async() => {
+        arrowDown(buttons[0]);
+        await onceOpened(subMenu);
         item = subMenu.$.overlay.querySelector('vaadin-context-menu-item');
+        expect(item).to.be.ok;
+        await nextRender(subMenu);
         arrowUp(item);
         await nextRender(subMenu);
         expect(subMenu.opened).to.be.false;
       });
 
+      (isIOS ? it.skip : it)('should switch menubar item and open submenu on arrow left', async() => {
```

Comment:
When pressing <kbd>Left</kbd> on a sub-menu, if the left button does not have children, the focus remains on the original button, and the focus-ring is lost entirely. I'd expect the focus to move to the left button regardless if it has children or not.

If I press <kbd>Right</kbd> on a non-parent sub-menu item, and the next button does not have children, it still gets the focus, like I'd expect.

(added comment on this line instead of a general comment just to get threads)

Reply:
Fixed. Had to use `stopImmediatePropagation()` for <kbd>Left</kbd> to stop handling in context-menu.